src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1133,20 +1132,64 @@ public override int Execute(DocumentsOperationContext context)
                                             if (_incoming._log.IsInfoEnabled)
                                                 _incoming._log.Info($"Conflict check resolved to Conflict operation, resolving conflict for doc = {item.Id}, with change vector = {item.ChangeVector}");
-                                            // if the conflict is going to ber resolved locally, that means that we have local work to do
+
+                                            if (removeClusterTx)
+                                            {
+                                                // if we are in a conflict status and the cluster tx flag should be removed, means that 
+                                                // we should try to resolve it to the document that have the higher group id etag
+
+                                                var groupId = _incoming._parent.Database.DatabaseGroupId;
+                                                var local = ChangeVectorUtils.GetEtagById(conflictingVector, groupId);
+                                                var remote = ChangeVectorUtils.GetEtagById(rcvdChangeVector, groupId);
+
+                                                if (local > remote)
+                                                {
+                                                    goto case ConflictStatus.AlreadyMerged;
+                                                }
+
+                                                if (remote > local)
+                                                {
+                                                    goto case ConflictStatus.Update;
+                                                }
+                                                // if local and remote are equal we continue to resolve it in the traditional way
+                                            }
+                                            
+                                            // if the conflict is going to be resolved locally, that means that we have local work to do
                                             // that we need to distribute to our siblings
                                             IsIncomingReplication = false;
-                                            _incoming._conflictManager.HandleConflictForDocument(context, item.Id, item.Collection, item.LastModifiedTicks, document, rcvdChangeVector, conflictingVector, item.Flags);
+                                            _incoming._conflictManager.HandleConflictForDocument(context, item.Id, item.Collection, item.LastModifiedTicks, document,
+                                                rcvdChangeVector, conflictingVector, item.Flags);
                                             break;
                                         case ConflictStatus.AlreadyMerged:
-                                            //nothing to do
+                                            if (removeClusterTx)
```

Comment:
What is the scenario where this actually happens, and what are the effects?

Reply:
We can get here from the conflict status. When the local groupId is higher than the remote one.
The scenario that I can think of: an external replication of cluster transaction to database that also performed cluster transaction.

So at first will resolve it to the local document. (and stripping the FromCluster flag)